File: 013/cpp/main.cpp
class Solution {
public:
    int getChar(char c) {
        switch (c) {
            case 'I': return 1;
            case 'V': return 5;
            case 'X': return 10;
            case 'L': return 50;
            case 'C': return 100;
            case 'D': return 500;
            default:  return 1000;
        }
    }

    int romanToInt(string s) {
        int v0 = 0;
        int v1 = 0;
        int r = 0;
        for (int i = s.size() - 1; i > -1; i--) {
            char c = s[i];
            v1 = getChar(c);
            if (v1 < v0) {
                r -= v1;
            } else {
                r += v1;
            }
            v0 = v1;
        }
        return r;
    }
};

File: 013/rust/main.rs
impl Solution {
    pub fn get_char(c: char) -> i32 {
        if c == 'I' {
            return 1;
        }
        if c == 'V' {
            return 5;
        }
        if c == 'X' {
            return 10;
        }
        if c == 'L' {
            return 50;
        }
        if c == 'C' {
            return 100;
        }
        if c == 'D' {
            return 500;
        }
        return 1000;
    }

    pub fn roman_to_int(s: String) -> i32 {
        let mut v0 = 0;
        let mut v1 = 0;
        let mut r = 0;
        let s = s.as_bytes();
        for i in 0..s.len() {
            let c = s[s.len() - 1 - i] as char;
            v1 = Solution::get_char(c);
            if (v1 < v0) {
                r -= v1;
            } else {
                r += v1;
            }
            v0 = v1;
        }
        return r;
    }
}
